### PR TITLE
Use ruff for pyupgrade and isort

### PR DESCRIPTION
**Pull request by zariiii9003**
You could activate many more [rules](https://docs.astral.sh/ruff/rules/) to be checked on commit and in CI.

### CODE REVIEW

File: src/asammdf/blocks/conversion_utils.py [hunk 1]
```diff
@@ -379,7 +378,7 @@ def from_dict(conversion: dict[str, Any]) -> v4b.ChannelConversion:
     return conversion
 
 
-def to_dict(conversion: ChannelConversionType) -> Union[dict, None]:
+def to_dict(conversion: ChannelConversionType) -> dict | None:
```
Comment (danielhrisca) [inline src/asammdf/blocks/conversion_utils.py:381]: Doesn't this require Python >= 3.10 ?

Comment (zariiii9003) [reply src/asammdf/blocks/conversion_utils.py:381]: Yes it does, but ruff ignores it, because the file contains `from __future__ import annotations`.

Comment (danielhrisca) [reply src/asammdf/blocks/conversion_utils.py:381]: Ok, let me know when it is ready for merging

Comment (zariiii9003) [reply src/asammdf/blocks/conversion_utils.py:381]: I'd say it's ready. This little change resulted in 73 modified files, that's enough for one PR 😄

**Comment by danielhrisca** [comment]
Thanks!

**Pull request merged**